docs/setup/deploying-a-runtime.md

```diff
@@ -0,0 +1,294 @@
+
+```
+oasis-node registry entity list -a $ADDR -v
+```
+
+Should give output similar to:
+
+<!-- markdownlint-disable line-length -->
+```
+{"v":1,"id":"JTUtHd4XYQjh//e6eYU7Pa/XMFG88WE+jixvceIfWrk=","nodes":["LQu4ZtFg8OJ0MC4M4QMeUR7Is6Xt4A/CW+PK/7TPiH0="]}
+{"v":1,"id":"+MJpnSTzc11dNI5emMa+asCJH5cxBiBCcpbYE4XBdso="}
+{"v":1,"id":"TqUyj5Q+9vZtqu10yw6Zw7HEX3Ywe0JQA9vHyzY47TU=","allow_entity_signed_nodes":true}
+```
+<!-- markdownlint-enable line-length -->
+
+In following steps we will register and run the [simple-keyvalue] runtime on the
+network.
+
+[simple-keyvalue]: ../../tests/runtimes/simple-keyvalue
+
+## Initializing a runtime
+
+<!--
+TODO: until https://github.com/oasisprotocol/oasis-core/issues/3093 is fixed
+this step needs to happen after the compute node is already started and ready
+(see later steps).
+-->
+
+To generate and sign a runtime registration transaction that will initialize and
+register the runtime we will use the `registry runtime gen_register` command.
+When initializing a runtime we need to specify various runtime parameters. To
+list all of the available parameters run: `registry runtime gen_register --help`
+subcommand.
+
+```
+oasis-node registry runtime gen_register --help
+```
+
+For additional information about runtimes and parameters see the
+[runtime documentation] and [code reference].
+
+Before generating the registration transaction, gather the following data and
+set up environment variables to simplify instructions.
+
+- `ENTITY_DIR` - Path to the entity directory created when starting the
+development network. This entity will be the runtime owner. The genesis used in
+the provisioning initial network step funds the `entity-2` entity (located
+within the net-runner directory), so that is the one we use in following
+instructions.
+- `GENESIS_JSON` - Path to the genesis.json file used in the development
+network.
+- `RUNTIME_ID` - See [runtime identifiers] on how to choose a runtime
+identifier. In the example we use
+ `8000000000000000000000000000000000000000000000000000000001234567`.
+- `NONCE` - Entity account nonce. If you followed the guide, nonce `0`
+would be the initial nonce to use for the entity. Note: make sure to keep
+updating the nonce when generating new transactions.
+
+```
+export ENTITY_DIR=/tmp/runtime-example/net-runner/network/entity-2/
+export GENESIS_JSON=/tmp/runtime-example/net-runner/network/genesis.json
+export RUNTIME_ID=8000000000000000000000000000000000000000000000000000000001234567
+export NONCE=0
+```
+
+[runtime identifiers]: ../runtime/identifiers.md
+
+```
+oasis-node registry runtime gen_register \
+  --transaction.fee.gas 1000 \
+  --transaction.fee.amount 1000 \
+  --transaction.file /tmp/runtime-example/register_runtime.tx \
+  --transaction.nonce $NONCE \
+  --genesis.file $GENESIS_JSON \
+  --signer.backend file \
+  --signer.dir $ENTITY_DIR \
+  --runtime.id $RUNTIME_ID \
+  --runtime.kind compute \
+  --runtime.executor.group_size 1 \
```

Comment:
yeah i don't know, keeping a list of all parameters (and explanations) would probably be a bit hard to keep up to date - it would also probably be just a copy of help strings, so might be good enough to just link there:
https://github.com/oasisprotocol/oasis-core/blob/3263a4dbbb190c410cc42fd0b15a36c813b58412/go/oasis-node/cmd/registry/runtime/runtime.go#L541-L585